src/options.cpp

```diff
@@ -69,6 +69,18 @@ class arg_parser {
     return matches;
   }
 
+  bool match_flag_option(std::string_view option_name) noexcept {
```

Comment:
Nit: I'd be lazy and delegate to the overload on line 59. Option parsing isn't performance-critical.

What you have now is perfectly fine too.